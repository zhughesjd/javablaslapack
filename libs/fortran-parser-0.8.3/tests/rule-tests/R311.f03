! Testing intrinsic-operator, R311
interface operator(.unaryop.)
end interface operator(.binaryop.)
interface operator(**)
end interface operator(//)
end
