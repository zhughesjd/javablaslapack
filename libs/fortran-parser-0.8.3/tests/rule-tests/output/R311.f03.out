R1101:main-program__begin:
R1203:interface-stmt__begin:
R 311:defined-operator: definedOp=[@6,54:62='.unaryop.',<189>,2:19] isExtendedIntrinsicOp=false
R1207:generic_spec: keyword=[@4,45:52='operator',<130>,2:10] name=null type=1501
R1203:interface-stmt: abstract_token=null keyword=[@2,35:43='interface',<118>,2:0] eos=[@8,64:64='\n',<4>,2:29] hasGenericSpec=true
R 311:defined-operator: definedOp=[@15,88:97='.binaryop.',<189>,3:23] isExtendedIntrinsicOp=false
R1207:generic_spec: keyword=[@13,79:86='operator',<130>,3:14] name=null type=1501
R1204:end-interface-stmt: keyword1=[@9,65:67='end',<184>,3:0] keyword2=[@11,69:77='interface',<118>,3:4] eos=[@17,99:99='\n',<4>,3:34] hasGenericSpec=true
R1203:interface-stmt__begin:
R 311:defined-operator: definedOp=[@22,119:120='**',<38>,4:19] isExtendedIntrinsicOp=true
R1207:generic_spec: keyword=[@20,110:117='operator',<130>,4:10] name=null type=1501
R1203:interface-stmt: abstract_token=null keyword=[@18,100:108='interface',<118>,4:0] eos=[@24,122:122='\n',<4>,4:22] hasGenericSpec=true
R 311:defined-operator: definedOp=[@31,146:147='//',<41>,5:23] isExtendedIntrinsicOp=true
R1207:generic_spec: keyword=[@29,137:144='operator',<130>,5:14] name=null type=1501
R1204:end-interface-stmt: keyword1=[@25,123:125='end',<184>,5:0] keyword2=[@27,127:135='interface',<118>,5:4] eos=[@33,149:149='\n',<4>,5:26] hasGenericSpec=true
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=2
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=false hasInternalSubprogramPart=false
